examples/perfmanager/main.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This example program shows how the `view` and `property` packages can
+be used to navigate a vSphere inventory structure using govmomi.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vmware/govmomi/examples"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/performance"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func main() {
+	examples.Run(func(ctx context.Context, c *vim25.Client) error {
+		// Find virtual machine list
+		f := find.NewFinder(c, true)
+
+		dc, err := f.DefaultDatacenter(ctx)
+		if err != nil {
+			return err
+		}
+
+		f.SetDatacenter(dc)
+
+		vms, err := f.VirtualMachineList(ctx, "*")
+		if err != nil {
+			return err
+		}
+
+		// Convert virtual machines into list of references
+		var vmsRefs []types.ManagedObjectReference
```

Comment:
I think using `ContainerView` instead of `Finder` would be better for this example:

* `ContainerView.Find` already returns `[]types.ManagedObjectReference`

* The `kind` param to `Find` could be changed from "VirtualMachine" to another object type (such as "HostSystem") and it looks like your example would still work.  What do you think?

```go
		m := view.NewManager(c)

		v, err := m.CreateContainerView(ctx, c.ServiceContent.RootFolder, nil, true)
		if err != nil {
			return err
		}

		defer v.Destroy(ctx)

		refs, err := v.Find(ctx, []string{"VirtualMachine"}, nil)
		if err != nil {
			return err
		}
```

Reply:
Great, much better. I didn't know that. :)